pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -226,53 +226,55 @@ func handleSyncOpResponse(ctx *downloaderContext, config types.DownloaderConfig,
 	status *types.DownloaderStatus, locFilename string,
 	key string, errStr string) {
 
+	// have finished the download operation
+	// based on the result, perform some storage
+	// management also
+
 	if status.ObjType == "" {
 		log.Fatalf("handleSyncOpResponse: No ObjType for %s\n",
 			status.ImageID)
 	}
+
 	locDirname := types.DownloadDirname + "/" + status.ObjType
 	if errStr != "" {
-		// Delete file
+		// Delete file, and update the storage
 		doDelete(ctx, key, locDirname, status)
-		status.PendingAdd = false
-		status.Size = 0
-		status.LastErr = errStr
-		status.LastErrTime = time.Now()
-		status.RetryCount += 1
+		// free the reserved storage
+		unreserveSpace(ctx, status)
+		status.HandleDownloadFail(errStr)
 		publishDownloaderStatus(ctx, status)
-		log.Errorf("handleSyncOpResponse failed for %s, <%s>\n",
+		log.Errorf("handleSyncOpResponse(%s): failed for with %s\n",
```

Comment:
"failed for with" should presumably be "failed with" or "failed: "

Reply:
Sure, done.